libs/store_helper/kernel/ebpf_registry_helper.c

```diff
@@ -9,11 +9,56 @@
 #include "ebpf_registry_helper.h"
 #include "ebpf_windows.h"
 
+#define EBPF_STORE_TAG 'OTSE'
+
 #define _EBPF_RESULT(x) (NT_SUCCESS(x) ? EBPF_SUCCESS : EBPF_FAILED)
 
 ebpf_store_key_t ebpf_store_root_key = NULL;
 const wchar_t* ebpf_store_root_sub_key = EBPF_ROOT_REGISTRY_PATH;
 
+wchar_t*
+ebpf_get_wstring_from_string(_In_ const char* text)
+{
+    NTSTATUS status;
+    ANSI_STRING ansi_string;
+    UNICODE_STRING unicode_string = {0};
+    uint32_t new_size;
+
+    RtlInitAnsiString(&ansi_string, text);
```

Comment:
Just checking - are all callers guaranteed to pass non-NULL? Alternatively, are these functions all safe with NULL input?

Reply:
(Applies to the other implementation below too).

Reply:
So the SAL annotation on `ebpf_get_wstring_from_string` says `_In_`, which should mean non-NULL. Since the caller is also a driver, there is no trust boundary, so I think we dont need to check for NULL.